src/modules/Nfs.ycp

```diff
@@ -352,6 +354,89 @@
 	return true;
     }
 
+    typedef string vertex_t;
+
+    vertex_t VertexFromEntry(map<string, any> e) {
+      return String::PadZeros(tostring(e["i"]:0), 6) + e["file"]:"";
+    }
+
+    string PathFromVertex(vertex_t v) {
+      return substring(v, 6);
+    }
+
+    /*
+     * DependsOn(a, b) is true if a must be mounted before b.
+     * That is, "/any" DependsOn "/", "/foo/bar" DependsOn "/foo",
+     * but "/foof" not DependsOn "/foo"
+     */
+    boolean DependsOn(vertex_t a, vertex_t b) {
+      a = PathFromVertex(a);
+      b = PathFromVertex(b);
+
+      if (a == b) return false;
+
+      boolean StartsWith(string s, string head) {
+        return substring(s, 0, size(head)) == head;
+      }
+
+      boolean EndsWith(string s, string tail) {
+        return substring(s, size(s) - size(tail), size(tail)) == tail;
+      }
+
+      if (! EndsWith(a, "/")) {
+        a = a + "/";
+      }
+      if (! EndsWith(b, "/")) {
+        b = b + "/";
+      }
+      return StartsWith(a, b);
+    }
+
+    map<vertex_t, list<vertex_t> > DependencyGraph(list<map<string, any> > fstab) {
+      typedef map<string, any> entry_t;
+
+      list<vertex_t> vertices = maplist(entry_t e, fstab,
+                                        ``( VertexFromEntry(e) ));
+
+      // the edges are in the "supports" direction, /foo -> /foo/bar
+      return listmap(entry_t e, fstab, {
+          vertex_t vertex = VertexFromEntry(e);
+          list<vertex_t> targets = filter(vertex_t t, vertices,
+                                          ``( DependsOn(t, vertex) ));
+          return $[vertex : targets];
+        });
+    }
+
+    /**
+     * Merge two partial fstabs so that they are properly ordered.
+     * For example / must go before anything else,
+     * /foo/bar must go before /foo/bar/baz
+     */
+    list<map<string, any> > MergeFstabs(list<map<string, any> > a,
+                                        list<map<string, any> > b) {
+      typedef map<string, any> entry_t;
+
+      integer i = 0;
+      list<entry_t> all = maplist(entry_t e, a + b, {
+          i = i + 1;
+          e = add(e, "i", i);
+          return e;
+      });
+
+      map<vertex_t, list<vertex_t> > supports = DependencyGraph(all);
+      list<list<vertex_t> > ordered = TopologicalSort(supports);
+      if (ordered[1]:[] != []) {
+        y2error("Oops, a cycle in fstab? %1", ordered[1]:[]);
+        //FIXME: bail out? use a plain all
+        return all;
+      }
+
+      map<vertex_t, entry_t> by_vertex = listmap(entry_t e, all,
+                                                 ``( $[ VertexFromEntry(e) : remove(e, "i")] ));
+
+      return maplist(string v, ordered[0]:[], ``( by_vertex[v]:$[] ));
```

Comment:
To be honest whole this method is not much readable. Too high count of WTF per line.